rfcs/0180-broken-package-removal.md

```diff
@@ -0,0 +1,99 @@
+---
+feature: broken-package-removal
+start-date: 2024-07-14
+author: Jörg "Mic92" Thalheim
+co-authors:
+shepherd-team: "@emilazy @preisi @jopejoe1"
+shepherd-leader: "@jopejoe1"
+related-issues:
+---
+
+# Summary
+
+[summary]: #summary
+
+This RFC defines conditions under which we remove broken packages or unmaintained leaf packages.
+The RFC does not aim to define all possible reasons for removing a package but instead focuses on simple, automatable rules.
+Since this document serves as a guideline for automatic package removal, it also allows for the possibility of removing a package earlier for other reasons.
+
+# Motivation
+
+[motivation]: #motivation
+
+Broken and unmaintained packages still consume valuable time.
+Broken packages need to be evaluated to some extent, and unmaintained packages might still need to be rebuilt when one of their dependencies changes.
+Removing unmaintained or unused code should improve overall package quality and help save resources.
+Since adding packages is easy, removing them should also be simple, ideally automated.
+
+# Detailed Design
+
+[design]: #detailed-design
+
+All broken packages or unmaintained leaf packages will be removed after a full NixOS release cycle.
+For example, if `hello-unmaintained` is marked as unmaintained in 23.11, it can be removed from master after the 24.05 release.
+
+If a NixOS module depends on any removed package and is non-functional without it, the module will be removed as well.
+In `pkgs/top-level/aliases.nix`, we can link to the pull request that removed the package, making it easier for users to recover the nix expression if needed.
+
+In the pull request that removes a package, we will also ping the people who have modified or updated the package, excluding those who only touched the package as part of a tree-wide update.
```

Comment:
Nit: "we will" is odd wording here. 

```suggestion
In the pull request that removes a package, the people who have substantially contributed the package declaration before should be pinged.
```

Reply:
Perhaps we should also limit this to people who are still active:

```suggestion
In the pull request that removes a package, the people who have substantially contributed to the package declaration before and are still active in the project should be pinged.
```

Reply:
OTOH, you do want the potentially inactive (it's often unknown what exactly is the case) to be pinged.

What I'm concerned with is people being pinged who have left the project long ago and don't care for packages to be removed they have at one point touched.